src/blob.rs

```diff
@@ -15,6 +17,17 @@ struct Header {
     flags: u64,
 }
 
+impl Blob {
+    pub fn from_file<P: AsRef<Path>>(_path: P) -> Result<Self, ()> {
+        // @TODO implement
+        unimplemented!()
+    }
+
+    pub fn is_active(&self) -> bool {
+        unimplemented!()
+    }
+}
+
 #[test]
```

Comment:
Is is ok to write tests in the file with implementation?

Reply:
You puts unit tests in each file in _src_ , with `#[cfg(test)]` annotation
Integration tests are entirely external to library and locates in _tests_ dir at the top level of project
Src: [Rust Book Test organization](https://doc.rust-lang.org/book/ch11-03-test-organization.html)

Reply:
>  The convention is to create a module named tests in each file to contain the test functions and to annotate the module with cfg(test).

According to Rust Book you should define a separate module.